src/pages/Product-listing/Product.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import {useState,useEffect} from 'react'
+import { Filter, Navbar } from '../../components'
+// import{Link} from 'react-router-dom';
+import axios from 'axios';
+import { useProduct } from '../../context';
+import './Product.css';
+import { getCategoryProduct, getRatingProduct, getSortedProduct } from '../../utils';
+import { getMetalCategoryProduct } from '../../utils/getCategoryProduct';
+
+
+const Product = () => {
+  const [productItem,setProductItem] = useState([]);
+  const {productState,productDispatch} = useProduct(); 
+  const metalCategorizedProduct = getMetalCategoryProduct(
+    productItem,
+    productState.metal.gold,
+    productState.metal.silver,
+    productState.metal.platinum) 
+  const categorizedProduct = getCategoryProduct(
+    metalCategorizedProduct,
+    productState.category.ring,
+    productState.category.earring,
+    productState.category.necklace,
+    productState.category.bangles)
+  const ratedProduct = getRatingProduct(categorizedProduct,productState);
+  const sortedProduct = getSortedProduct(ratedProduct,productState);
+
+  useEffect(()=>{
+    (
+      async () => {
+        try{
+            const {data:{products}} = await axios.get('/api/products')
+            setProductItem(products)
+        } catch(e){
+            console.log(e);
```

Comment:
you may change `console.log(e)` here to `console.error(e)` till you are not handling the errors